css/geometry-1/DOMMatrixInit-validate-fixup.html

```diff
@@ -0,0 +1,185 @@
+  }
+  if ('d' in dict && 'm22' in dict) {
+    dict.d = dict.m22;
+  }
+  if ('e' in dict && 'm41' in dict) {
+    dict.e = dict.m41;
+  }
+  if ('f' in dict && 'm42' in dict) {
+    dict.f = dict.m42;
+  }
+}
+
+// https://drafts.fxtf.org/geometry/#create-a-dommatrix-from-the-dictionary
+// The spec will use only the 6 relevant values to create a 2d matrix, so the test
+// needs this normalization for e.g. {m13: -0, is2D: true} when comparing `m13`.
+function normalizeDict2D(dict) {
+  delete dict.m13;
+  delete dict.m14;
+  delete dict.m23;
+  delete dict.m24;
+  delete dict.m31;
+  delete dict.m32;
+  delete dict.m33;
+  delete dict.m34;
+  delete dict.m43;
+  delete dict.m44;
+}
+
+[
+  {a: 1, m11: 2},
+  {b: 0, m12: -1},
+  {c: Infinity, m21: -Infinity},
+  {d: 0, m22: NaN},
+  {e: 1, m41: 1.00000001},
+  {f: 0, m42: Number.MIN_VALUE},
+  {m13: 1, is2D: true},
+  {m14: 1, is2D: true},
+  {m23: 1, is2D: true},
+  {m24: 1, is2D: true},
+  {m31: 1, is2D: true},
+  {m32: 1, is2D: true},
+  {m33: 0, is2D: true},
+  {m33: -0, is2D: true},
+  {m33: -1, is2D: true},
+  {m34: 1, is2D: true},
+  {m43: 1, is2D: true},
+  {m44: 0, is2D: true},
+].forEach(dict => {
+  test(() => {
+    assert_true('fromMatrix' in DOMMatrix, 'fromMatrix should exist');
+    assert_throws(new TypeError(), () => DOMMatrix.fromMatrix(dict));
+  }, `${format_dict(dict)} (invalid)`);
+});
+
+[
+  {},
+  {is2D: undefined}, // undefined means the dictionary member is not present
+  {a: 1, m11: 1},
+  {b: 0, m12: undefined},
+  {c: 0, m21: 0},
+  {c: 0, m21: -0},
+  {c: -0, m21: 0},
+  {c: -0, m21: -0},
+  {d: Infinity, m22: Infinity},
+  {e: -Infinity, m41: -Infinity},
+  {f: NaN, m42: NaN},
+  {f: NaN, m42: NaN, is2D: true},
+  {f: 0, m42: null}, // null is converted to 0
+  {f: -0, m42: null}, // null is converted to 0
+  {a: 2},
+  {b: 2},
+  {c: 2},
+  {d: 2},
+  {e: 2},
+  {f: 2},
+  {a: -0, b: -0, c: -0, d: -0, e: -0, f: -0},
+  {a: -0, b: -0, c: -0, d: -0, e: -0, f: -0, is2D: true},
+  {m11:2},
+  {m12:2},
+  {m21:2},
+  {m22:2},
+  {m41:2},
+  {m42:2},
+  {m11: -0, m12: -0, m21: -0, m22: -0, m41: -0, m42: -0},
+  {m11: -0, m12: -0, m21: -0, m22: -0, m41: -0, m42: -0, is2D: true},
+  {m13: 0, is2D: true},
+  {m13: -0, is2D: true},
+  {m14: 0, is2D: true},
+  {m14: -0, is2D: true},
+  {m23: 0, is2D: true},
+  {m23: -0, is2D: true},
+  {m24: 0, is2D: true},
+  {m24: -0, is2D: true},
+  {m31: 0, is2D: true},
+  {m31: -0, is2D: true},
+  {m32: 0, is2D: true},
+  {m32: -0, is2D: true},
+  {m33: 1, is2D: true},
+  {m34: 0, is2D: true},
+  {m34: -0, is2D: true},
+  {m43: 0, is2D: true},
+  {m43: -0, is2D: true},
+  {m44: 1, is2D: true},
+  {is2D: true},
+].forEach(dict => {
+  test(() => {
+    const matrix = DOMMatrix.fromMatrix(dict);
+    normalizeDictForMinusZero(dict);
+    normalizeDict2D(dict);
+    for (let prop in dict) {
+      if (dict[prop] !== undefined && dict[prop] !== null) {
+        assert_equals(matrix[prop], dict[prop], prop);
+      }
```

Comment:
Shouldn't you test all values? E.g. I would think normalizeDict2D would set all non-2D members to 0, so that you could test equality there.

Same below. In general skipping properties seems bad for test coverage :)